infrastructure/builder/src/generate/generators/docs-crawler.js

```diff
@@ -0,0 +1,131 @@
+
+    // This function will be called for each URL to crawl.
+    // Here you can write the Puppeteer scripts you are familiar with,
+    // with the exception that browsers and pages are automatically managed by the Apify SDK.
+    // The function accepts a single parameter, which is an object with the following fields:
+    // - request: an instance of the Request class with information such as URL and HTTP method
+    // - page: Puppeteer's Page object (see https://pptr.dev/#show=api-class-page)
+    handlePageFunction: async ({ request, page }) => {
+      // A function to be evaluated by Puppeteer within the browser context.
+      const pageFunction = (posts) => {
+        const data = [];
+
+        posts.forEach((post) => {
+          const makeData = (type) => {
+            const pageTitleElement = post.querySelector('h1');
+            const title = pageTitleElement && pageTitleElement.innerText.trim();
+            const links = Array.from(post.querySelectorAll(`${type} .anchor-link-style`));
+
+            return links
+              .map(link => {
+                const { parentElement } = link;
+                const content = parentElement && parentElement.innerText.trim();
+                const href = link && link.href;
+
+                if (!content || !href) {
+                  return null;
+                }
+
+                return {
+                  title,
+                  type,
+                  content,
+                  href,
+                };
+              })
+              .filter(Boolean);
+          };
+
+          // Extract headers
+          // For each header, grab the page title and its link
+          const result = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].map(makeData).flat();
+
+          if (result.length) {
+            data.push(result);
+          }
+        });
+
+        return data;
+      };
+      // Processing page
+      const data = await page.$$eval(PAGE_CONTENT_SELECTOR, pageFunction);
+
+      // Store asset.
+      await Apify.pushData(data);
+
+      // Find a _relative_ link to the next page and enqueue it if it exists.
+      await Apify.utils.enqueueLinks({
+        page,
+        requestQueue,
+        selector: LINK_SELECTOR,
+      });
+    },
+  });
+}
+
+async function readCrawlerAssets() {
+  const filenames = await readdir(join(REPO_ROOT, TEMP_STORAGE_PATH));
+  const files = await Promise.all(
+    filenames.map((filename =>
+      readJSON(join(TEMP_STORAGE_PATH, filename))
+    )));
+
+  // Flatten the data
+  return [].concat(...files);
+}
+
+function removeApifyStorage() {
+  return rimraf(join(REPO_ROOT, GENERATED, APIFY_STORAGE));
+}
+
+exports.tasks = [{
+  title: 'Crawler that indexes the docs site',
+  provides: ['docs-crawler'],
+  run: async (requirements, utils) => {
+    // Prepare a clean state for the crawler
+    await removeApifyStorage();
+
+    const crawler = await apifyCrawler({ url: 'https://taskcluster-web.netlify.com/docs' });
```

Comment:
The `url` should probably not be hard-coded. This value should correspond to the domain it's being deployed against. Is there an environment variable already set up that has the domain?

Reply:
There's no deployment yet -- this is generating code that goes back into the repository, after all.

Also, running a headless browser adds a *really* big and potentially fragile requirement to our contributors for running `yarn generate`.  There's also the potential for different browsers or different versions of those browsers to produce different output, which would lead to unnecessary index changes getting checked in, or spurious failures of `yarn generate` in CI.

That said, to get this working I think you'll need to run tc-ui within the `yarn generate` process, and index the documents from there.